src/controllers/userController.ts

```diff
@@ -1,125 +1,212 @@
-}
+};
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const updateUser = async(req: Request, res: Response, next: NextFunction) => {
-  const id = Number(req.params.uid)
-
-  const password = req.body.password;
-  const email = req.body.email;
-  await User.update({ password: password, email: email }, {
-    where: {
-      id: id
+const getUserbyId = async (req: Request, res: Response, next: NextFunction) => {
+    const id = Number(req.params.uid) || 0;
+    const user = await User.findOne({
+        where: { id: id },
+    });
+    if (user) {
+        res.status(200).json({ user: user });
+    } else {
+        res.status(404).json({ message: "User not found" });
     }
-  });
-  
-}
+};
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const postLogin = async(req: Request, res: Response, next: NextFunction) => {
-  const email = req.body.email;
-  const password = req.body.password;
-   await User.findOne({
-    where: {email: email}
-  })
-  .then (user => {
-    if (!user) {
-     return res.status(403).json({"message": "user not found", "email": email})
-          }
-          if (!!user) {
-          bcrypt.compare(password, user.password )
-            .then(result => {
-                      if (result && process.env.JWT_SECRET ) {
-                        
-                 const token = jwt.sign({email: user.email}, process.env.JWT_SECRET, {expiresIn: '8h'} )       
-               return res.status(200).json({"message": "OK", token: token})
-              } 
-              else {
-               return res.status(403).json({"message": "wrong password", "email": email})
-              }
-                    
-            })
-          }
-    
-    }).catch(err => {
-      console.log('general error')
-      console.log(err)
-    })
+const deleteUser = async (req: Request, res: Response, next: NextFunction) => {
+    //We are checking if user has a profile. If so, profile is being deleted and then the user
 
-  
-  } 
-  
+    const id = Number(req.params.uid);
 
+    try {
+        const profile = await Profile.findOne({
+            where: { user: id },
+        });
+        const user = await User.findOne({
+            where: { id },
+        });
+
+        if (profile) {
+            await Profile.destroy({
+                where: {
+                    user: id,
+                },
+            });
+        }
+
+        if (user) {
+            await User.destroy({
+                where: {
+                    id: id,
+                },
+            });
+            res.status(200).json({ message: "User deleted" });
+        } else {
+            res.status(400).json({ message: "User doesn't exist" });
+        }
+    } catch (err) {
+        console.log(err);
+
+        res.status(500).json({ message: "error deleting user" });
+    }
+};
 
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const updateUser = async (req: Request, res: Response, next: NextFunction) => {
```

Comment:
For user we do only login and register for now.
You can leave this method as it is, but it will be changed later.